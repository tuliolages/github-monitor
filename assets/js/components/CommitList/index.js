import React from 'react';
import PropTypes from 'prop-types';

const CommitList = (props) => {
  const {commits,updateFilters} = props;
  return (
    <div>
      {commits.length !== 0 && (
        <div>
          <div className="card card-outline-secondary my-4">
            <div className="card-header">
              Commit List
            </div>

            <div className="card-body">
              {commits.map((commit, index) => (
                <div key={commit.sha}>
                  <div className="avatar">
                    <img alt={commit.author} className="img-author" src={commit.avatar} />
                  </div>
                  <div className="commit-details">
                    <p>
                      {commit.message}
                    </p>
                    <small className="text-muted">
                      <span className="filter-link" onClick={() => updateFilters({author: commit.author})}>{commit.author}</span>
                      {' '}
                      authored
                      {' '}
                      on
                      {' '}
                      <span className="filter-link" onClick={() => updateFilters({repository: commit.repository.id})}>{commit.repository.name}</span>
                      {' '}
                      at
                      {' '}
                      {commit.date}
                    </small>
                    {index !== commits.length - 1 && <hr />}
                  </div>
                </div>
              ))}
            </div>
          </div>
        </div>
      )}
    </div>
  );
};

CommitList.propTypes = {
  commits: PropTypes.arrayOf(PropTypes.object).isRequired,
  updateFilters: PropTypes.func.isRequired,
};

export default CommitList;
