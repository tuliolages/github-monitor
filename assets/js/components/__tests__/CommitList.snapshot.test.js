import React from 'react';
import { shallow } from 'enzyme';
import CommitList from '../CommitList';

test('Render filter selects', () => {
  const component = shallow(
    <CommitList
      commits={[{
        message: 'add github_external_api tests',
        sha: '4f9ff7c6e4d26dfc50d69c3ddbc49717928a9e92',
        author: 'unknown',
        url: 'https://api.github.com/repos/tuliolages/github-monitor/commits/4f9ff7c6e4d26dfc50d69c3ddbc49717928a9e92',
        avatar: '',
        date: '2021-02-06T20:13:56-03:00',
        repository: {
          name: 'github-monitor',
          id: 13,
        },
      },
      {
        message: 'creates initial view tests and fixtures',
        sha: '65dfa477eefd6a4c98ae781740b042b86a454673',
        author: 'unknown',
        url: 'https://api.github.com/repos/tuliolages/github-monitor/commits/65dfa477eefd6a4c98ae781740b042b86a454673',
        avatar: '',
        date: '2021-02-06T17:04:29-03:00',
        repository: {
          name: 'github-monitor',
          id: 13,
        },
      }]}
      updateFilters={() => { }}
    />
  );
  expect(component).toMatchSnapshot();
});

test('Render empty filter selects', () => {
  const component = shallow(
    <CommitList
      commits={[]}
      updateFilters={() => { }}
    />
  );
  expect(component).toMatchSnapshot();
});
