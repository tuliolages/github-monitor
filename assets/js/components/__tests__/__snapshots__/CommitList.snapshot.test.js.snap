// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render empty filter selects 1`] = `
<div>
  <div>
    <div
      className="card card-outline-secondary my-4"
    >
      <div
        className="card-header"
      >
        Commit List
      </div>
      <div
        className="card-body"
      >
        <p>
          There are no commits
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Render filter selects 1`] = `
<div>
  <div>
    <div
      className="card card-outline-secondary my-4"
    >
      <div
        className="card-header"
      >
        Commit List
      </div>
      <div
        className="card-body"
      >
        <div
          key="4f9ff7c6e4d26dfc50d69c3ddbc49717928a9e92"
        >
          <div
            className="avatar"
          >
            <img
              alt="unknown"
              className="img-author"
              src=""
            />
          </div>
          <div
            className="commit-details"
          >
            <p>
              add github_external_api tests
            </p>
            <small
              className="text-muted"
            >
              <span
                className="filter-link"
                onClick={[Function]}
              >
                unknown
              </span>
               
              authored
               
              on
               
              <span
                className="filter-link"
                onClick={[Function]}
              >
                github-monitor
              </span>
               
              at
               
              2021-02-06T20:13:56-03:00
            </small>
            <hr />
          </div>
        </div>
        <div
          key="65dfa477eefd6a4c98ae781740b042b86a454673"
        >
          <div
            className="avatar"
          >
            <img
              alt="unknown"
              className="img-author"
              src=""
            />
          </div>
          <div
            className="commit-details"
          >
            <p>
              creates initial view tests and fixtures
            </p>
            <small
              className="text-muted"
            >
              <span
                className="filter-link"
                onClick={[Function]}
              >
                unknown
              </span>
               
              authored
               
              on
               
              <span
                className="filter-link"
                onClick={[Function]}
              >
                github-monitor
              </span>
               
              at
               
              2021-02-06T17:04:29-03:00
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
